hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreScanner.java

```diff
@@ -757,7 +757,7 @@ public boolean next(List<Cell> outResult, ScannerContext scannerContext) throws
         }
       } while ((cell = this.heap.peek()) != null);
 
-      if (count > 0) {
+      if (count > 0 || (storeLimit > -1 && this.countPerRow > (storeLimit + storeOffset))) {
```

Comment:
We do not need to add this extra check any more?

Reply:
No, I just forget to delete this.